modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/v2_2_1/DEDataDefinitionFieldLinkUpgradeProcess.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.data.engine.internal.upgrade.v2_2_1;
+
+import com.liferay.data.engine.internal.upgrade.v2_1_1.util.DEDataDefinitionFieldLinkTable;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Lino Alves
+ */
+public class DEDataDefinitionFieldLinkUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		alter(
+			DEDataDefinitionFieldLinkTable.class,
+			new AlterColumnType("fieldName", "VARCHAR(255) null"));
```

Comment:
@linolaoj, sorry for not mention it in the previous review, but, apparently, it's a good upgrade practice check if the column exists before trying to manipulate it, please check [here](https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Choose-the-proper-upgrade-step).